### PR TITLE
Refactor OpenReport API call for Report GetHistory

### PR DESCRIPTION
**Pull request by sketchydroide**
CC @marcaaron @Beamanator 

### Fixed Issues
https://github.com/Expensify/Expensify/issues/212870

### Tests
This needs to be tested along with the Web PR https://github.com/Expensify/Web-Expensify/pull/34429 (now merged)

1. Make sure to have a chat with a few messages, I would say around the 100, that you already have opened before.
2. Open the app and head to the chat in question
3. Verify that it loads the report and shows the messages. Verify as well that no loader is shown
---
1. Make sure to have a third chat with a less than 50 messages, I would say around the 30, that you already have opened before.
2. Open the app and head to the chat in question
3. Verify that it loads the report and shows the messages. Verify as well that no loader is shown
---
1. Make sure to have a second chat with around the same amount of messages, that you already have  NOT opened before.
2. Open the app and head to the chat in question
3. Verify a loader is shown
5. Verify that after the report is loaded the report appears and shows the messages.
---
- [ ] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [ ] I linked the correct issue in the `### Fixed Issues` section above
- [ ] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I added steps for local testing in the `Tests` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

- [ ] Verify that no errors appear in the JS console

### CODE REVIEW

File: src/libs/actions/Report.js
Comment (marcaaron) [inline src/libs/actions/Report.js:1050]: This seems like something the view should handle. We can still set the loading state. The view will decide to use it or not based on whether it has initial report actions to display.

File: src/pages/home/report/ReportActionsView.js
Comment (marcaaron) [inline src/pages/home/report/ReportActionsView.js:148]: All of the `updateNewMarkerAndMarkReadOnce()` stuff can go now. That was only existing to support a race condition where the `maxSequenceNumber` is not up to date because the report object had not loaded yet. Since we are relying on the report actions in the backend we can get rid of it.

File: src/pages/home/report/ReportActionsView.js
Comment (marcaaron) [inline src/pages/home/report/ReportActionsView.js:204]: I think we still want this.

**Review by marcaaron — CHANGES_REQUESTED** [review]

**Review by tgolen — CHANGES_REQUESTED** [review]
@sketchydroide Would you mind adding a better title and description to this PR? I am looking at it with very little context from the original issue and I'm confused about what exactly this PR is meant to be doing. The issue is long, it mentions several different pieces, so I'm missing the context of the changes. It also needs QA tests to be added.

**Comment by sketchydroide** [comment]
Sorry @tgolen this is still WIP, I always forget that this Repo adds a reviewer as soon as I create it

I will add all the things, also I'll improve the title

**Comment by tgolen** [comment]
OK, thanks! Creating the PR in draft mode is a good practice where you can avoid that issue. 👍

File: src/pages/home/report/ReportActionsView.js
Comment (sketchydroide) [reply src/pages/home/report/ReportActionsView.js:204]: so just remove the fetchData bellow?
I tought it was weird to have a openReport in case is fully visible, and fetch data if it's not

Comment (marcaaron) [reply src/pages/home/report/ReportActionsView.js:204]: I see. Well, with this removed we would be marking the report as read if it's not in view. I'm now unsure if `OpenReport` is the best command name to use in this case then. 

If sometimes we want to fetch the report actions + mark the report as read and other times we just want to fetch the report actions - then these seem like two separate actions that we should have two separate commands for as per the 1:1:1 API design philosophy.

Maybe it is best to leave the `fetchData()` call as is for now and we can do something like this in a follow up:

- Create a `ReconnectToReport` command
- Have this command fetch the latest history without marking the report as read

Eventually we would change this to

```js
if (this.getIsReportFullyVisible()) {
    Report.openReport(this.props.reportID);
} else {
    Report.reconnect(this.props.reportID);
}
```

Comment (sketchydroide) [reply src/pages/home/report/ReportActionsView.js:204]: should I create a GH for this?
I'll get fetchData back in 😅

File: src/pages/home/ReportScreen.js
Comment (sketchydroide) [inline src/pages/home/ReportScreen.js:143]: The linter asked me to do a return early here.

File: src/pages/home/ReportScreen.js
Comment (sketchydroide) [inline src/pages/home/ReportScreen.js:166]: @marc suggested this and I agreed, this makes it easier to understand the purpose of the check

File: src/pages/home/report/ReportActionsView.js
Comment (sketchydroide) [reply src/pages/home/report/ReportActionsView.js:139]: removed all of them

File: src/pages/home/ReportScreen.js
Comment (marcaaron) [reply src/pages/home/ReportScreen.js:143]: heads up there's an open pr where this stuff is changing :D https://github.com/Expensify/App/pull/10208

File: src/pages/home/report/ReportActionsView.js
Comment (marcaaron) [reply src/pages/home/report/ReportActionsView.js:204]: Yeah I think we will need a follow up for this.

Comment (marcaaron) [reply src/pages/home/report/ReportActionsView.js:204]: Mostly since there is no version of "fetch data" that doesn't mark the report as read now.

**Comment by marcaaron** [comment]
Code is looking good. So, we just need QA tests added now?

File: src/pages/home/ReportScreen.js
Comment (sketchydroide) [reply src/pages/home/ReportScreen.js:145]: yeah I merged that one already

**Comment by marcaaron** [comment]
Looks like this one is ready for review, but should still be on HOLD until the Web-Expensify PR is on production.

File: src/pages/home/ReportScreen.js
Comment (marcaaron) [inline src/pages/home/ReportScreen.js:145]: Was this just old code that should have been removed in a previous PR?

File: src/libs/actions/Report.js
Comment (marcaaron) [inline src/libs/actions/Report.js:1052]: This feels like a `write()` to me as we are setting the last read sequence number in the command and will want to make sure that happens when we come back from offline.

File: src/ONYXKEYS.js
Comment (tgolen) [inline src/ONYXKEYS.js:104]: What's the difference between this collection, the one above it, and the one below it? This feels like it's getting a little bit out of hand.

Comment (tgolen) [reply src/ONYXKEYS.js:104]: Why are we using collections for these things at all?

**Review by marcaaron — CHANGES_REQUESTED** [review]
Changes LGTM mainly have a question about whether `OpenReport` should be a `write()` (I think it should).

File: src/pages/home/ReportScreen.js
Comment (tgolen) [reply src/pages/home/ReportScreen.js:165]: Can you please update the method description in the JSDocs to be more clear? It is answering "what", but not "why"

Comment (tgolen) [reply src/pages/home/ReportScreen.js:165]: Shouldn't this really be more like `isLoadingReportActions`? It doesn't look like it has anything to do with "initial" actions or not. I'm not even sure what "initial" implies.

**Review by tgolen — CHANGES_REQUESTED** [review]

**Comment by marcaaron** [comment]
While testing I noticed that `OpenReport` is getting called 3 times 🤔 

https://user-images.githubusercontent.com/32969087/183712792-119e66b0-5e41-43a8-bff6-1d9b78840b2e.mp4

Unrelated to this PR though as I am seeing similar behavior on production.

File: src/ONYXKEYS.js
Comment (marcaaron) [reply src/ONYXKEYS.js:104]: I think we can remove the `IS_LOADING_INITIAL_REPORT_ACTIONS` one. It seems like nothing is subscribing to this key. So that just leaves the others...

But I _think_ we are using collections because: 

- an individual report could be "loading" or "loading more" while not in view and we don't want to incorrectly show a report as loading when it's not or not loading when it is
- there wasn't any other alternative proposed

Any specific concern with this approach? 

Thinking on what we should we do as an alternative... One idea is to possibly use the `report` object to store a `isLoadingActions` and `isLoadingMoreActions`. The report object is a little bloated already with all sorts of random stuff, but maybe it's better than what we have here.

File: src/pages/home/ReportScreen.js
Comment (marcaaron) [reply src/pages/home/ReportScreen.js:165]: "initial" means you have no `reportActions` at all to display vs. having some `reportActions` to display, but still in the process of loading the next set of actions ones. There are two different animations displayed based on which situation we have.

File: src/ONYXKEYS.js
Comment (tgolen) [reply src/ONYXKEYS.js:104]: > One idea is to possibly use the report object to store a isLoadingActions and isLoadingMoreActions.

Yeah, this is basically what I would propose otherwise. It sounds like that's the best practice considering pattern B.

My concern is that I think this leads to an anti-pattern. It becomes a bunch of keys holding a single boolean value, when the boolean value could be a property on the normal report collection. If a new engineer looks at this, and sees it's OK, then what are the next collections with single values? Collections for just report names? You can use this argument to say that instead of one collection with `n` reports, then it's OK to have `n` collections for `n` reports (one collection for each report property).

Unless there is a real advantage to having these kinds of simple value collections (I am just not seeing it), I think we should avoid it and stick to our best practices. (not trying to blame anyone!)

Comment (marcaaron) [reply src/ONYXKEYS.js:104]: Wanna move this to Slack to discuss? After, we can put it down in the style guide / fix anything that is not following the best practice.

Comment (sketchydroide) [reply src/ONYXKEYS.js:104]: I will remove the `isLoadingInitialReportActions_`

File: src/libs/actions/Report.js
Comment (sketchydroide) [reply src/libs/actions/Report.js:1050]: you are right it should be a write

File: src/pages/home/ReportScreen.js
Comment (sketchydroide) [reply src/pages/home/ReportScreen.js:145]: hmm, this might have been related to the PR that was merged in, but this was not removed???
from what I remember it was to remove the loader after a certain time, but we don't use it anymore.
I know I removed the timer in this [commit](https://github.com/Expensify/App/pull/10164/commits/b301b28689ccbd4dd7fb3073ad03927e4b599443)

File: src/pages/home/ReportScreen.js
Comment (sketchydroide) [reply src/pages/home/ReportScreen.js:166]: are you happy with that @tgolen or do you want me to rename the var?

**Comment by sketchydroide** [comment]
>While testing I noticed that OpenReport is getting called 3 times 🤔

Maybe we can open a GH to check on this, that feels like the kind of thing we should look into, not sure if it's the scope of this PR though

File: src/pages/home/ReportScreen.js
Comment (tgolen) [reply src/pages/home/ReportScreen.js:166]: I really like Marc's explanation. Could you add that as a code comment to help explain it to others who will see this? If the comment is added, then I think the name is OK then.

Comment (sketchydroide) [reply src/pages/home/ReportScreen.js:166]: Will do

File: src/pages/home/ReportScreen.js
Comment (marcaaron) [inline src/pages/home/ReportScreen.js:297]: I guess one downside of storing this on the `report` will be that we cannot use `initWithStoredValues: false` 🤔 

Though, I'm not sure if it's necessary to do this.

File: src/ONYXKEYS.js
Comment (marcaaron) [reply src/ONYXKEYS.js:103]: Ok seems like we had a good discussion [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1660150371779569) and are adding this to the style guide. 

So, @sketchydroide we just need to update these so that the boolean properties are on the report instead of collection keys.

File: src/pages/home/ReportScreen.js
Comment (sketchydroide) [reply src/pages/home/ReportScreen.js]: I added a shorter version of Marcs answer, that I feel makes more sense to read, let me know if that works.

File: src/ONYXKEYS.js
Comment (sketchydroide) [reply src/ONYXKEYS.js]: I think I have done this for the `isLoadingReportActions`
Should I do it for the IS_LOADING_MORE_REPORT_ACTIONS as well in this PR as well, it's from the previous PR for `ReadOldestAction`

Comment (sketchydroide) [reply src/ONYXKEYS.js]: I'm not 100% I did that correctly, but I think so, did I miss something, it feels odd that `isLoadingReportActions` is not always defined...

File: src/pages/home/ReportScreen.js
Comment (marcaaron) [inline src/pages/home/ReportScreen.js:285]: Something looks off here and seems we are passing `withWindowDimensions` as a property of the `withOnyx` mapping now.

we probably want something like this:

```js
compose(
    withNetwork(), 
    withWindowDimensions,
    withOnyx({
```

File: src/pages/home/report/ReportActionsView.js
Comment (marcaaron) [inline src/pages/home/report/ReportActionsView.js:426]:
```suggestion
```

File: src/pages/home/report/ReportActionsView.js
Comment (marcaaron) [inline src/pages/home/report/ReportActionsView.js:429]: We are already passing the report as a prop to this component so we don't need `withOnyx` anymore

https://github.com/Expensify/App/blob/972b83549e10cf1f05310f269f07a2a83b1e2253/src/pages/home/ReportScreen.js#L242-L245

File: src/ONYXKEYS.js
Comment (marcaaron) [reply src/ONYXKEYS.js]: Looks correct to me. 

>it feels odd that isLoadingReportActions is not always defined...

This is JavaScript which means 99% of all things are possibly `undefined`, but hey at least it's a boolean!

File: src/libs/actions/Report.js
Comment (marcaaron) [reply src/libs/actions/Report.js:990]: 👍

**Review by marcaaron — COMMENTED** [review]
It's looking good @sketchydroide ! spotted a few other things.

File: src/pages/home/ReportScreen.js
Comment (sketchydroide) [reply src/pages/home/ReportScreen.js:285]: migtht have been missed with a merge, adding it

**Review by Beamanator — COMMENTED** [review]
A few little suggestions, didn't have time to test today

**Comment by sketchydroide** [comment]
https://github.com/Expensify/Web-Expensify/pull/34429 has been merged, removing the hold from the title

File: src/libs/actions/Report.js
Comment (sketchydroide) [reply src/libs/actions/Report.js:1005]: yep, it was a multiple entry there, but now one of them was removed and I forgot to edit it.

It seems something likt linter should pick up though.

File: src/pages/home/ReportScreen.js
Comment (sketchydroide) [reply src/pages/home/ReportScreen.js:311]: possibly, I don't remember edit this (although possible)

File: src/libs/actions/Report.js
Comment (tgolen) [inline src/libs/actions/Report.js:653]: Now that this is all this method does, could you please just remove the method `fetchInitialActions()` and call `fetchActions()` directly?

File: src/libs/actions/Report.js
Comment (tgolen) [inline src/libs/actions/Report.js:1029]: NAB hahaha... we're `write`ing an action called `Read`? That doesn't make a whole lot of sense. The API command should probably be something more like `MarkOldestActionAsRead`

File: src/pages/home/report/ReportActionsView.js
Comment (tgolen) [inline src/pages/home/report/ReportActionsView.js:456]: How does the component get `this.props.report` now without using `withOnyx()`?

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/libs/actions/Report.js
Comment (marcaaron) [reply src/libs/actions/Report.js:1029]: If this is a NAB - I prefer the current name and think it's OK. The alternative suggestion confuses me a bit because I am imagining the user manually marking something (and this is something that happens as a side-effect of the user reading something with their eyes and not "reading data from the server").

File: src/pages/home/report/ReportActionsView.js
Comment (marcaaron) [reply src/pages/home/report/ReportActionsView.js:456]: from the parent component here:

https://github.com/Expensify/App/blob/972b83549e10cf1f05310f269f07a2a83b1e2253/src/pages/home/ReportScreen.js#L242-L245

File: src/libs/actions/Report.js
Comment (marcaaron) [reply src/libs/actions/Report.js:653]: Just a note that it's going to be removed entirely in the next round of PRs.

**Comment by marcaaron** [comment]
@sketchydroide I think this still needs to be on HOLD as the Web-E PR has been merged but not yet deployed.

File: src/libs/actions/Report.js
Comment (marcaaron) [reply src/libs/actions/Report.js:990]: Please make this a `write()`

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionsView.js
Comment (tgolen) [reply src/pages/home/report/ReportActionsView.js:456]: Oh, Thanks! I don't think we should pass both the `reportID` and `report`. Those are redundant. Let's just pass `report` and get the `reportID` off that.

File: src/libs/actions/Report.js
Comment (Beamanator) [reply src/libs/actions/Report.js:1029]: > (and this is something that happens as a side-effect of the user reading something with their eyes and not "reading data from the server").

+1 - Since the action is the "reading" I think it's fine, but maybe we could go with `ViewOldestAction` so we don't use the term "read"

File: src/libs/actions/Report.js
Comment (sketchydroide) [reply src/libs/actions/Report.js:653]: I can still replace it though, it's not that much work I think

File: src/libs/actions/Report.js
Comment (sketchydroide) [reply src/libs/actions/Report.js:1029]: 😅  I think this is still a read, and I just changed the wrong method, I meant to change OpenReport to Write

File: src/pages/home/report/ReportActionsView.js
Comment (sketchydroide) [reply src/pages/home/report/ReportActionsView.js:456]: I can do that, but reportID at the moment is `required` not sure if we should make the `report` required as well,  not sure if we should have a default value for reportID

Comment (marcaaron) [reply src/pages/home/report/ReportActionsView.js:456]: This sounds like an easy change and a good thing to look into. I also think it would be OK to do in a follow up and not add additional refactoring scope to these changes. I don't immediately see a clear reason for passing the `reportID` and `report` as separate props, but I haven't looked into it deeply enough to say there isn't one.

**Review by marcaaron — DISMISSED** [review]
LGTM! Thanks for the changes. We still need to wait for [this Web-Expensify PR](https://github.com/Expensify/Web-Expensify/pull/34429) to go to production before merging.

**Comment by marcaaron** [comment]
Web-E PR is now on production.

File: src/pages/home/report/ReportActionsView.js
Comment (tgolen) [reply src/pages/home/report/ReportActionsView.js:456]: I think we should make `report` the required prop then, but as always, make sure you take a look at the code and understand it first before making a change like that.

I won't fight tooth-and-nail about it being done in this PR, but I also think this is a refactor PR and I am not too concerned about scope for something small like this.

Comment (sketchydroide) [reply src/pages/home/report/ReportActionsView.js:456]: just made the change and tested it after, seems to work great, let me know if you are happy with that @tgolen

File: src/pages/home/report/ReportActionsView.js
Comment (tgolen) [inline src/pages/home/report/ReportActionsView.js:39]: Can you also make this one required?
```suggestion
        reportID: PropTypes.number.isRequired,
```

I think it's the only property that is required for a report.

**Review by tgolen — CHANGES_REQUESTED** [review]
Looks great now! Just this one last small change I noticed

**Review by marcaaron — APPROVED** [review]

**Review by tgolen — APPROVED** [review]

**Comment by sketchydroide** [comment]
the tests seems to be failing but it does not seem related to the last commit

**Comment by sketchydroide** [comment]
merging main seems o have fixed it

**Comment by sketchydroide** [comment]
@Beamanator is OOO, but his comments were addresed, so I think we are good

**Pull request merged**
